server/base/src/main/java/org/apache/accumulo/server/init/Initialize.java

```diff
@@ -748,24 +735,27 @@ private String getRootUserName(SiteConfiguration siteConfig, Opts opts) throws I
     if (opts.cliPassword != null) {
       return opts.cliPassword.getBytes(UTF_8);
     }
-    String rootpass;
-    String confirmpass;
+    char[] rootpass;
+    char[] confirmpass;
+    String strrootpass;
+    String strconfirmpass;
     do {
-      rootpass = getLineReader().readLine(
-          "Enter initial password for " + rootUser + getInitialPasswordWarning(siteConfig), '*');
+      rootpass = System.console().readPassword(
+          "Enter initial password for " + rootUser + getInitialPasswordWarning(siteConfig));
       if (rootpass == null) {
         System.exit(0);
       }
-      confirmpass =
-          getLineReader().readLine("Confirm initial password for " + rootUser + ": ", '*');
+      confirmpass = System.console().readPassword("Confirm initial password for " + rootUser + ":");
       if (confirmpass == null) {
         System.exit(0);
       }
-      if (!rootpass.equals(confirmpass)) {
+      strrootpass = new String(rootpass);
+      strconfirmpass = new String(confirmpass);
+      if (!strrootpass.equals(strconfirmpass)) {
         log.error("Passwords do not match");
       }
     } while (!rootpass.equals(confirmpass));
```

Comment:
This should probably be the comparison of the string version otherwise, you would need Arrays.,equals

Reply:
I have already fixed that